lib/src/app/feature/google_authenticaiton/data/repositories/authenticaiton_repository.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:firebase_auth/firebase_auth.dart';
+
+import '../../models/authentication_detail.dart';
+import '../domain/authentication_firebase_provider.dart';
+import '../domain/google_sign_in_provider.dart';
+
+class AuthenticationRepository {
```

Comment:
Can you please create abstract repository?
Please take a reference from home feature.